joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreator.scala

```diff
@@ -83,12 +83,36 @@ class AstCreator(val relPathFileName: String, val parserResult: ParserResult)(im
     nodeInfo.node match {
       case GenDecl          => astForGenDecl(nodeInfo)
       case FuncDecl         => astForFuncDecl(nodeInfo)
-      case _: BasePrimitive => Seq(astForPrimitive(nodeInfo))
+      case _: BasePrimitive => astForPrimitive(nodeInfo)
       case _: BaseExpr      => astsForExpression(nodeInfo)
       case _: BaseStmt      => astsForStatement(nodeInfo)
       case _                => Seq()
     }
   }
+
+  protected def astForEmptyArrayInitializer(primitive: ParserNodeInfo): Ast = {
+    val arrayType: String = primitive.node match {
+      case ArrayType =>
+        s"${primitive.json(ParserKeys.Elt)(ParserKeys.Name)}[]"
+      case CompositeLit =>
+        s"${primitive.json(ParserKeys.Type)(ParserKeys.Elt)(ParserKeys.Name)}[]"
+      case _ =>
+        Defines.empty
+    }
+
+    Ast(
+      callNode(
+        primitive,
+        primitive.code,
+        Operators.arrayInitializer,
+        Operators.arrayInitializer,
+        DispatchTypes.STATIC_DISPATCH,
+        Option(Defines.empty),
+        Option(arrayType.replaceAll("\"", "")) // The "" around the typename is eliminated
```

Comment:
Does this only happen for array types? If it happens elsewhere, then we may want to handle this in a single place

Reply:
@DavidBakerEffendi Identifying a Type needs to be handled in more general way. So that same function can be used at all the places, where Type Identification is required.

@karan-batavia we will syncup on this tomorrow.

Reply:
@DavidBakerEffendi I am working on handling TypeFullName identification at a central place. I need to refactor some code for that purpose. I will be handling that part in my next PR.